src/components/Matrix/TableRow/index.js

```diff
@@ -2,64 +2,42 @@ import { useState } from "react";
 import CancelIcon from "@mui/icons-material/Cancel";
 import { TableCell } from "../TableCell";
 import { CustomButton } from "../../shared/CustomButton";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  CLEAR_CLOSEST_ID,
+  INCREASE_AMOUNT,
+  REMOVE_ROW,
+  SET_CLOSEST_ID,
+} from "../../../reducer/actions/matrix.actions";
+
+export const TableRow = ({ row, index, styles }) => {
+  const dispatch = useDispatch();
 
-export const TableRow = ({
-  row,
-  closestCellsCount,
-  index,
-  styles,
-  matrix,
-  onSetData,
-  closestCellIds,
-  onSetClosestCellIds,
-}) => {
   const [isHoverSum, setIsHoverSum] = useState(false);
 
+  const { matrix, closestCellIds } = useSelector((state) => state);
+
   const sumRow = row.reduce((sum, current) => sum + current.amount, 0);
 
   const removeMatrixRow = () =>
-    onSetData(matrix.filter((item) => matrix.indexOf(item) !== index));
-
-  const getPersentOfCell = (item) => Math.round((100 * item.amount) / sumRow);
-
-  const findClosestIds = (data) => {
-    const closestIds = matrix
-      .flat()
-      .filter((item) => data.id !== item.id)
-      .sort(
-        (a, b) =>
-          Math.abs(a.amount - data.amount) - Math.abs(b.amount - data.amount)
-      )
-      .slice(0, closestCellsCount)
-      .map((item) => item.id);
-    onSetClosestCellIds(closestIds);
-  };
-  
-  const handleIncreaseAmount = (cell) =>
-    matrix.map((row) =>
-      row.map((item) => {
-        const increase = () => {
-          item.amount += 1;
-          return item;
-        };
-        return item.id === cell.id ? increase() : item;
-      })
+    dispatch(
+      REMOVE_ROW(matrix.filter((item) => matrix.indexOf(item) !== index))
```

Comment:
why are your functions in uppercase? it should name just removeMatrixRow(index). should pass as argument only index . the filter will be a cutter: matrix.filter((_, index) => index !== deleteRowIndex)